awx/main/models/credential/__init__.py

```diff
@@ -653,15 +698,15 @@ class TowerNamespace:
         # build a normal namespace with secret values decrypted (for
         # ansible-playbook) and a safe namespace with secret values hidden (for
         # DB storage)
-        for field_name, value in credential.inputs.items():
+        injectable_fields = list(credential.inputs.keys()) + credential.dynamic_input_fields
+        for field_name in list(set(injectable_fields)):
```

Comment:
If a field has a value assigned to it normally, it can be found in `credential.inpputs`. If it's sourced from an external credential, it'll be in `credential.dynamic_input_fields`. We need to include both when injecting values.